packages/curve-ui-kit/src/features/connect-wallet/lib/ConnectionContext.tsx

```diff
@@ -118,34 +123,35 @@ export const ConnectionProvider = <
         if (walletChainId && walletChainId !== chainId) {
```

Comment:
We need to check if the document is visible to avoid fighting between tabs

```suggestion
        if (walletChainId && walletChainId !== chainId && !document.hidden) {
```